File: test/four/app/src/main/java/com/z3d/frame3d/dis/FrameBuildSprite.java
package com.z3d.frame3d.dis;

import android.util.Log;

import com.z3d.base.ObjData;
import com.z3d.core.Context3D;
import com.z3d.display.Display3DSprite;
import com.z3d.frame3d.FrameNodeVo;
import com.z3d.program.Shader3D;
import com.z3d.scene.Scene3D;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class FrameBuildSprite  extends FrameBaseDisplay {
    private static final String TAG ="FrameBuildSprite" ;
    public FrameBuildSprite(Scene3D val) {
        super(val);
        this.scene3D.progrmaManager.registe(FrameBuildShader.FrameBuildShader,new FrameBuildShader(scene3D));
        this.shader3D=this.scene3D.progrmaManager.getProgram(FrameBuildShader.FrameBuildShader);
    }
    @Override
    public void setFrameNodeUrl(FrameNodeVo $vo) {
        super.setFrameNodeUrl($vo);
        this.groupItem = new ArrayList<>();
        Display3DSprite $dis = new Display3DSprite(this.scene3D);
        $dis.setObjUrl($vo.resurl);
        try {
            JSONObject objInfo=(JSONObject)$vo.materialInfoArr.get(0);
            $dis.setPicUrl(objInfo.getString("url"));
        } catch (JSONException e) {

        }

        this.groupItem.add($dis);
    }
    @Override
    public void upData() {
        super.upData();
       if( this.sceneVisible){
           for (int i=0;i<this.groupItem.size();i++){
               this.drawTempDisplay(this.groupItem.get(i));
           }
       }

    }
    private void drawTempDisplay(Display3DSprite display) {
        if(display.objData==null){
            return;
        }
        ObjData objData= display.objData;

        Log.d(TAG, "drawTempDisplay: ");
        Context3D ctx=this.scene3D.context3D;
        ctx.setProgame(this.shader3D.program);
        this.setVc();
        ctx.setVa(this.shader3D,"pos",3,objData.vertexBuffer);
        ctx.setVa(this.shader3D,"v2Uv",2,objData.uvBuffer);
        ctx.setRenderTexture(this.shader3D,"fs0",display.baseTextureRes.textTureInt,0);
        ctx.drawCall(display.objData.indexBuffer,objData.treNum);

    }

    protected void setVc()
    {
        Context3D ctx=this.scene3D.context3D;
        ctx.setVcMatrix4fv(this.shader3D, Shader3D.vpMatrix3D,this.scene3D.camera3D.modelMatrix.m);
        ctx.setVcMatrix4fv(this.shader3D,"posMatrix3D",this.posMatrix3d.m);

    }


}

File: metalios/iosgl/iosgl/pan3d/frame3d/dis/FrameBuildSprite.m
//
//  FrameBuildSprite.m
//  iosgl
//
//  Created by pan3dme on 2020/11/16.
//  Copyright © 2020 zhao. All rights reserved.
//
#import "FrameBuildSprite.h"
#import "Scene3D.h"
#import "FrameBuildShader.h"
#import "ProgrmaManager.h"
#import "Context3D.h"
@implementation FrameBuildSprite

- (instancetype)init:(Scene3D *)val
{
    self =[super init:val];
    [self inidShader];
    return self;
}
-(void)inidShader
{
    [self.scene3D.progrmaManager registe:FrameBuildShader.shaderStr shader3d: [[FrameBuildShader alloc]init:self.scene3D]];
    self.shader3d=  [self.scene3D.progrmaManager getProgram:FrameBuildShader.shaderStr];
}

-(void)setFrameNodeUrl:(FrameNodeVo*)nodeVo;
{
    [super setFrameNodeUrl:nodeVo];
    FrameBuildSprite* this=self;
    this.groupItem =[[NSMutableArray alloc] init];
    Display3DSprite* dis =[[Display3DSprite alloc]init:this.scene3D];
    [dis setObjUrl:nodeVo.resurl];
    NSDictionary* info =nodeVo.materialInfoArr[0];
    [dis setPicUrl:  [info valueForKey:@"url"]];
    [this.groupItem addObject:dis];
    
}
- (void)upFrame
{
    FrameBuildSprite* this=self;
    if( this.sceneVisible){
        for (int i=0;i<this.groupItem.count;i++){
            [self drawTempDisplay:this.groupItem[i]];
        }
    }
}

-(void)drawTempDisplay:(Display3DSprite*)display
{
    
    if(display.objData==nil){
        return;
    }
    ObjData* objData= display.objData;
    FrameBuildSprite* this=self;
    
    NSLog(@"这里了这里了");
    
    /*
    
    Context3D *ctx=this.scene3D.context3D;
    GLuint progame= self.shader3d.program;
    glUseProgram(progame);
    
    [self setVc];
    [ctx pushVa:objData.verticesBuffer];
    [ctx setVaOffset:this.shader3d name:"vPosition" dataWidth:3 stride:0 offset:0];
    [ctx pushVa:objData.uvBuffer];
    [ctx setVaOffset:this.shader3d name:"texcoord" dataWidth:2 stride:0 offset:0];
    [ctx setRenderTexture:self.shader3d name:@"fs0"  texture:display.textureRes.textTureLuint level:0];
    [ctx drawCall: objData.indexBuffer  numTril:objData.trinum];
    */
    
}
- (void)setVc;
{
    FrameBuildSprite* this=self;
    /*
    Context3D *context3D=this.scene3D.context3D;
    Matrix3D* viewM=this.viewMatrix;
    [context3D setVcMatrix4fv:this.shader3d name:"vpMatrix3D" data:viewM.m];
    [context3D setVcMatrix4fv:this.shader3d name:"posMatrix3D" data:this.posMatrix3d.m];
     */
}
@end
